src/lib/Scenes/Artwork/Components/RequestConditionReport.tsx

```diff
@@ -0,0 +1,185 @@
+import { Button, Flex } from "@artsy/palette"
+import { RequestConditionReport_artwork } from "__generated__/RequestConditionReport_artwork.graphql"
+import { RequestConditionReport_me } from "__generated__/RequestConditionReport_me.graphql"
+import { RequestConditionReportMutation } from "__generated__/RequestConditionReportMutation.graphql"
+import { RequestConditionReportQuery } from "__generated__/RequestConditionReportQuery.graphql"
+import { Modal } from "lib/Components/Modal"
+import { defaultEnvironment } from "lib/relay/createEnvironment"
+import { Schema, track } from "lib/utils/track"
+import React from "react"
+import { View } from "react-native"
+import { commitMutation, createFragmentContainer, graphql, QueryRenderer, RelayProp } from "react-relay"
+import { PayloadError } from "relay-runtime"
+
+interface RequestConditionReportProps {
+  artwork: RequestConditionReport_artwork
+  me: RequestConditionReport_me
+  relay: RelayProp
+}
+
+interface State {
+  requestingConditionReport: boolean
+  showConditionReportRequestedModal: boolean
+  showErrorModal: boolean
+  errorModalText: string
+}
+
+@track({
+  action_name: Schema.ActionNames.RequestConditionReport,
+  context_module: Schema.ContextModules.ArtworkDetails,
+})
+export class RequestConditionReport extends React.Component<RequestConditionReportProps, State> {
+  state: State = {
+    requestingConditionReport: false,
+    showConditionReportRequestedModal: false,
+    showErrorModal: false,
+    errorModalText: "",
+  }
+
+  requestConditionReport = () => {
+    const { artwork, relay } = this.props
+    return new Promise(async (resolve, reject) => {
```

Comment:
@brainbicycle Is it possible to provide a type argument to the returned promise, [similar to the comparable code in Reaction](https://github.com/artsy/reaction/blob/master/src/Apps/Artwork/Components/ArtworkDetails/RequestConditionReport.tsx#L53)?

I believe that doing so means that the [`data` on L85](https://github.com/artsy/emission/pull/2050/files#diff-2ddabf7101faf8d9fc2cb4fdd4255203R85) could retain it type, leading to further type safety downstream

Reply:
Yes it is ✅